dotnet/src/Connectors/Connectors.GoogleVertexAI/Services/VertexAIGeminiTextGenerationService.cs

```diff
@@ -38,14 +40,17 @@ public sealed class VertexAIGeminiTextGenerationService : ITextGenerationService
     {
         Verify.NotNullOrWhiteSpace(model);
         Verify.NotNullOrWhiteSpace(bearerKey);
+        Verify.NotNullOrWhiteSpace(location);
+        Verify.NotNullOrWhiteSpace(projectId);
 
         this._textGenerationClient = new GeminiTextGenerationClient(new GeminiChatCompletionClient(
 #pragma warning disable CA2000
             httpClient: HttpClientProvider.GetHttpClient(httpClient),
 #pragma warning restore CA2000
             modelId: model,
             httpRequestFactory: new VertexAIHttpRequestFactory(bearerKey),
-            endpointProvider: new VertexAIEndpointProvider(new VertexAIConfiguration(location, projectId)),
+            chatGenerationEndpoint: new Uri($"https://{location}-aiplatform.googleapis.com/v1/projects/{projectId}/locations/{location}/publishers/google/models/{model}:generateContent"),
```

Comment:
Dont IOC the endpoints if they are using only for testing purposes, get the `location`, `projectId` and `model` as parameters to the client instead.

Reply:
Look this [commit](https://github.com/microsoft/semantic-kernel/pull/5075/commits/5410ef06ab9661f2b17755ebf06878e6dcf996c4) reference:

Reply:
@RogerBarreto this is not for testing purposes, but if I move creating endpoint to client class I also have to introduce parameter to client constructor (type: Vertex or Google) I can do it but I think current solution is more elegant

Reply:
For example OpenAI are passing endpoint https://github.com/microsoft/semantic-kernel/blob/21ce67608cd666ef630f81cc9fc91777471025a0/dotnet/src/Connectors/Connectors.OpenAI/ChatCompletion/AzureOpenAIChatCompletionService.cs#L33

 ale okay I feel I won't win so I will change it :)

Reply:
@Krzysztof318 Move the all clients to be internal as well as its constructors and the parameters can be endpoints for now, but even so I would recommend moving the knowledge of the endpoint to be internal of a client and not provided from outside.

You could have your Vertex/Google Client : GeminiClient for that.

Reply:
> For example OpenAI are passing endpoint

OpenAI endpoint actually is not the best name, its actually the deployment Endpoint/Base Url, the specific API endpoints are on top of the `BaseUri` endpoint provided, same for huggingface and `Ollama` connectors.

Reply:
Okay I didn't know that.
Fixed endpoints :)